streams/readable-byte-streams/tee.any.js

```diff
@@ -0,0 +1,776 @@
+
+}, 'ReadableStream teeing with byte source: canceling branch1 should not impact branch2');
+
+promise_test(async () => {
+
+  const rs = new ReadableStream({
+    type: 'bytes',
+    start(c) {
+      c.enqueue(new Uint8Array([0x01]));
+      c.enqueue(new Uint8Array([0x02]));
+      c.close();
+    }
+  });
+
+  const [branch1, branch2] = rs.tee();
+  branch2.cancel();
+
+  const [chunks1, chunks2] = await Promise.all([readableStreamToArray(branch1), readableStreamToArray(branch2)]);
+  assert_equals(chunks1.length, 2, 'branch1 should have two chunks');
+  assert_array_equals([...chunks1[0]], [0x01], 'first chunk from branch1 should be correct');
+  assert_array_equals([...chunks1[1]], [0x02], 'second chunk from branch1 should be correct');
+  assert_array_equals(chunks2, [], 'branch2 should have no chunks');
+
+}, 'ReadableStream teeing with byte source: canceling branch2 should not impact branch1');
+
+promise_test(async () => {
+
+  const reason1 = new Error('We\'re wanted men.');
+  const reason2 = new Error('I have the death sentence on twelve systems.');
+
+  let resolve;
+  const promise = new Promise(r => resolve = r);
+  const rs = new ReadableStream({
+    type: 'bytes',
+    cancel(reason) {
+      assert_array_equals(reason, [reason1, reason2],
+        'the cancel reason should be an array containing those from the branches');
+      resolve();
+    }
+  });
+
+  const [branch1, branch2] = rs.tee();
+  await Promise.all([
+    branch1.cancel(reason1),
+    branch2.cancel(reason2),
+    promise
+  ]);
+
+}, 'ReadableStream teeing with byte source: canceling both branches should aggregate the cancel reasons into an array');
+
+promise_test(async () => {
+
+  const reason1 = new Error('This little one\'s not worth the effort.');
+  const reason2 = new Error('Come, let me get you something.');
+
+  let resolve;
+  const promise = new Promise(r => resolve = r);
+  const rs = new ReadableStream({
+    type: 'bytes',
+    cancel(reason) {
+      assert_array_equals(reason, [reason1, reason2],
+        'the cancel reason should be an array containing those from the branches');
+      resolve();
+    }
+  });
+
+  const [branch1, branch2] = rs.tee();
+  await Promise.all([
+    branch2.cancel(reason2),
+    branch1.cancel(reason1),
+    promise
+  ]);
+
+}, 'ReadableStream teeing with byte source: canceling both branches in reverse order should aggregate the cancel reasons into an array');
+
+promise_test(async t => {
+
+  const theError = { name: 'I\'ll be careful.' };
+  const rs = new ReadableStream({
+    type: 'bytes',
+    cancel() {
+      throw theError;
+    }
+  });
+
+  const [branch1, branch2] = rs.tee();
+  await Promise.all([
+    promise_rejects_exactly(t, theError, branch1.cancel()),
+    promise_rejects_exactly(t, theError, branch2.cancel())
+  ]);
+
+}, 'ReadableStream teeing with byte source: failing to cancel the original stream should cause cancel() to reject on branches');
```

Comment:
For these do you think it'd be easier to parameterize over bytes/non-bytes in a single file?

Reply:
The order in which I wrote these non-bytes/bytes tests does not match the order in which they landed in the PRs. 😛

You're right, I'll see if I can make these tests parameterized. 🙂